src/annotator/bucket-bar.js

```diff
@@ -14,7 +14,7 @@ import { ListenerCollection } from './util/listener-collection';
 export default class BucketBar {
   /**
    * @param {HTMLElement} container
-   * @param {Pick<import('./guest').default, 'anchors'|'selectAnnotations'>} guest
+   * @param {Pick<import('./guest').default, 'anchors'|'scrollToAnchor'|'selectAnnotations'>} guest
```

Comment:
Just a detail curiosity, but can you help me understand why this kind of typing is being used here? Is it to be explicit that these are the only members of `guest` that this class' constructor/methods will make use of? Or do we anticipate different shapes of `guest` being provided?

Reply:
> Is it to be explicit that these are the only members of guest that this class' constructor/methods will make use of? 

Yes. I didn't write this originally but I think the intent was to limit the amount of coupling between the `Guest` and other code, which is wise. A more common way to do this would be to locally define a type/interface that matches/is compatible with a subset of the `Guest` class. Using `Pick` has the advantage that it is less code, but the downside that the picked fields might not be exactly the same as if you described exactly what the current code (`BucketBar` here).